Mono.Cecil.Cil/PortablePdb.cs

```diff
@@ -249,6 +249,12 @@ public void Dispose ()
 				return;
 
 			WritePdbHeap ();
+
+			if (module_metadata.string_heap != pdb_metadata.string_heap) {
```

Comment:
Can't `module_metadata` and `pdb_metadata` be the same? SetModuleMetadata has a check for that so I assumed it was possible. I moved the check to WriteTableHeap.